linera-views/src/common.rs

```diff
@@ -325,32 +325,41 @@ pub trait ReadableKeyValueStore<E> {
     async fn find_key_values_by_prefix(&self, key_prefix: &[u8]) -> Result<Self::KeyValues, E>;
 
     /// Reads a single `key` and deserializes the result if present.
-    async fn read_value<V: DeserializeOwned>(&self, key: &[u8]) -> Result<Option<V>, E>
+    fn read_value<V: DeserializeOwned>(
+        &self,
+        key: &[u8],
+    ) -> impl Future<Output = Result<Option<V>, E>>
     where
+        Self: Sync,
         E: From<bcs::Error>,
```

Comment:
We use `fn() -> impl Future` instead of `async fn()` here because default implementations of `async fn` confuse `trait_variant`, but it seems fine with default implementations of `-> impl Trait` `fn`s.

Reply:
Would it make sense to add a comment in the code?

Reply:
I think it should be okay since if someone tries to change this it will fail to compile.

Reply:
(and this applies to several places, so I'd have to copy-paste the comment several times in order to make sure the reader caught it)

Reply:
I think Mathieu Baudet meant adding a TODO(#XXX) that corresponds to considering the resolution of that bug. There are several pending issues here like the confusion of the trait and having a mention could help in dealing with that. The idea is that if something needs you 10 minutes to figure out then a comment is worthwhile.

Reply:
I didn't add an issue because we're currently blocked on the upstream bug — but I've added a comment above the default implementations to save future readers the time of figuring it out :)